### PR TITLE
#108 Open app from dock

### LINKED ISSUE #108: Reopen app from dock
On OS X, when closing app by tapping cross button app moving to a background but can't started again from dock.

### PR DESCRIPTION
**Pull request by G07cha**
Fixes #108 

Current implementation:
- On opening new window app will be showed only after web content will be loaded(prevent appearing white screen)
- On reopening app from dock(OS X) will be started new app
- Window size set to 700x450 as suggested @iiegor

[Screencast](https://vimeo.com/138723730)

### CODE REVIEW

**Comment by shockone** [comment]
I think we should remember the window size.

**Comment by RobertoUa** [comment]
@shockone yes remember the size by default. But also add a settings window where user can enable/disable it. And also add an option to "aways start with maximized window"

**Comment by shockone** [comment]
I guess settings should be a separate task.

**Comment by RobertoUa** [comment]
@shockone yes, of course.

**Comment by G07cha** [comment]
Will do.

**Comment by G07cha** [comment]
Done, also partial-implemented 'Start maximized' feature(OS X don't allow start applications in full screen by default as i know)

File: src/main/main.js [hunk 1]
```diff
@@ -7,19 +7,68 @@ process.env.PATH += ':/usr/local/bin';
...
-    mainWindow.focus();
-    menu.setMenu(app, mainWindow);
+app.on('activate-with-no-open-windows', function () {
+	mainWindow = createWindow();
 });
+
+function createWindow() {
+	var path = require("path");
+	var fs = require("fs");
+	var userPrefsPath = path.join(app.getDataPath(), "preferences.json");
```
Comment (shockone) [inline src/main/main.js]: We already have serialization setup. Could you try to use it? https://github.com/black-screen/black-screen/blob/master/src/Terminal.ts#L98

File: src/main/main.js [hunk 2]
```diff
@@ -7,19 +7,68 @@ process.env.PATH += ':/usr/local/bin';
...
+	menu.setMenu(app, mainWindow);
+	
+	window.on('close', function () {
+		//Remembering window size and position before exit
+		var data = {
+			bounds: mainWindow.getBounds(),
+			maximized: window.isMaximized()
+		};
+		
+		fs.writeFileSync(userPrefsPath, JSON.stringify(data));
```
Comment (shockone) [inline src/main/main.js]: I think we can do that on resize.

Comment (G07cha) [inline src/main/main.js]: Not necessary, if user closing window he expects that on opening next time it will have same size.

Comment (shockone) [inline src/main/main.js]: Okay.

File: src/main/main.js (hunk 1 above)
Comment (G07cha) [inline src/main/main.js]: I had no experience with typescript but I'll try.

**Comment by joshtoo** [comment]
@RobertoUa - All of the linting support for .js & .ts will be in shortly.

**Comment by shockone** [comment]
@G07cha, how about leaving the window dimensions untouched in this PR and dealing with them separately? After all, the title says "Open app from dock".

**Comment by G07cha** [comment]
> shockone
> I think we should remember the window size.

I just did what you said.

**Comment by shockone** [comment]
I still believe it's the right thing to do, but maybe we should separate these issues. Although, you decide.

**Comment by G07cha** [comment]
Merged commit with implemented reopening from dock functionality, 'Save window bounds' feature will be in separate PR, closing this one.

**Pull request closed without merging**